src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -249,13 +249,14 @@ public bool IsUnityProject()
             var globalProperties = new Dictionary<string, string>
             {
                 { PropertyNames.DesignTimeBuild, "true" },
+                { PropertyNames.BuildingInsideVisualStudio, "true" },
                 { PropertyNames.BuildProjectReferences, "false" },
                 { PropertyNames._ResolveReferenceDependencies, "true" },
                 { PropertyNames.SolutionDir, solutionDirectory + Path.DirectorySeparatorChar },
 
                 // This properties allow the design-time build to handle the Compile target without actually invoking the compiler.
                 // See https://github.com/dotnet/roslyn/pull/4604 for details.
-                { PropertyNames.ProvideCommandLineInvocation, "true" },
+                { PropertyNames.ProvideCommandLineArgs, "true" },
```

Comment:
💭 I'm somewhat surprised this is needed. Or was that a different issue?

Reply:
It's a different issue. Per my PR comment, there a couple of unrelated changes.